src/main/java/org/sunbird/searchby/service/SearchByService.java

```diff
@@ -468,4 +468,212 @@ private Map<String, Object> listCompetencyDetails(List<String> identifiers) thro
 	}
+	public SBApiResponse getCompetencyDetailsV1() {
+		SBApiResponse apiResponse = new SBApiResponse();
+		Map<String, Object> result = new HashMap<>();
+
+		try {
+			Map<String, Object> reqBody = prepareCompositeSearchRequestBody();
+			Map<String, Object> compositeSearchRes = outboundRequestHandlerService.fetchResultUsingPost(
+					cbExtServerProperties.getSbSearchServiceHost() + cbExtServerProperties.getSbCompositeV4Search(), reqBody, null);
+
+			if (compositeSearchRes == null || !compositeSearchRes.containsKey(Constants.RESULT)) {
+				throw new Exception("Composite search result is null or missing.");
+			}
+
+			Map<String, Object> compositeSearchResult = (Map<String, Object>) compositeSearchRes.get(Constants.RESULT);
+			List<Map<String, Object>> facetsList = (List<Map<String, Object>>) compositeSearchResult.get(Constants.FACETS);
+
+			logger.info("facetsList :: "+facetsList);
+			String url = cbExtServerProperties.getKnowledgeServiceUrl() + cbExtServerProperties.getFrameworkReadEndpoint() + cbExtServerProperties.getKcmFrameworkName();
+			logger.info("framework url:: "+url);
+
+			Map<String, Object> fracSearchRes = (Map<String, Object>) outboundRequestHandlerService.fetchResult(url);
+
+			if (fracSearchRes == null || !fracSearchRes.containsKey(Constants.RESULT)) {
+				throw new Exception("Framework read result is null or missing.");
+			}
+
+			Map<String, Object> fracResponseList = (Map<String, Object>) fracSearchRes.get(Constants.RESULT);
+			Map<String, Object> frameworkMap = (Map<String, Object>) fracResponseList.get(Constants.FRAMEWORK);
+			List<Map<String, Object>> categories = (List<Map<String, Object>>) frameworkMap.get(Constants.CATEGORIES);
+
+			// Validate and process the response
+			if (!CollectionUtils.isEmpty(facetsList) && !CollectionUtils.isEmpty(categories)) {
+				List<Map<String, Object>> contentList = processFacetData(facetsList, categories);
+				result.put(Constants.CONTENT, contentList);
+				apiResponse.setResult(result);
+				apiResponse.setResponseCode(HttpStatus.OK);
+			} else {
+				throw new Exception("Facets or Framework categories are empty.");
+			}
+
+		} catch (Exception ex) {
+			logger.error("Error while fetching competency details: ",ex);
+			apiResponse.setResponseCode(HttpStatus.INTERNAL_SERVER_ERROR);
+			apiResponse.getParams().setErr("Failed to get competency details.");
+			apiResponse.getParams().setErrmsg(ex.getMessage());
+		}
+
+		return apiResponse;
+	}
+
+	private Map<String, Object> prepareCompositeSearchRequestBody() {
+		Map<String, Object> reqBody = new HashMap<>();
+		Map<String, Object> req = new HashMap<>();
+		req.put(Constants.FACETS, Arrays.asList(
+				Constants.COMPETENCIES_V6_AREA_NAME,
+				Constants.COMPETENCIES_V6_THEME_NAME,
+				Constants.COMPETENCIES_V6_SUB_THEME_NAME
+		));
+		Map<String, Object> filters = new HashMap<>();
+		filters.put(Constants.PRIMARY_CATEGORY, Arrays.asList(Constants.COURSE, Constants.PROGRAM, Constants.CURATED_PROGRAM));
```

Comment:
there might be change required on key value pair of this... 
it is better to use -> courseCategory attribute and input should come from configuration